components/script/dom/htmllinkelement.rs

```diff
@@ -795,3 +1039,77 @@ impl PreInvoke for PrefetchContext {
         true
     }
 }
+
+struct PreloadContext {
+    /// The `<link>` element that caused this prefetch operation
+    link: Trusted<HTMLLinkElement>,
+
+    resource_timing: ResourceFetchTiming,
+
+    /// The url being prefetched
+    url: ServoUrl,
```

Comment:
nit: please change "prefetch" to "preload" in these two comments, to avoid confusion with `<link rel=prefetch>`.